lib/cli-engine.js

```diff
@@ -391,12 +392,14 @@ function getCacheFile(cacheFile, cwd) {
  */
 function CLIEngine(options) {
 
+    options = (options instanceof Object) ? assign(options) : {};
+    options.cwd = options.cwd || process.cwd();
+
     /**
      * Stored options for this instance
      * @type {Object}
      */
-    this.options = assign(Object.create(defaultOptions), options || {});
-
+    this.options = assign(defaultOptions, options);
```

Comment:
@gronke the first argument of `assign()` receives the properties from the other arguments, so you are overwriting `defaultOptions` each time this is called. 

`Object.create(default Options)` creates a new object that inherits from defaultOptions`, so it's a new copy each time. 

Unless you were seeing a bug with how this was, please revert this change. If you were saying a bug, please describe it.